tests/c/array.c

```diff
@@ -0,0 +1,35 @@
+//===--- array.c - test input file for iwyu -------------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// IWYU_ARGS: -I .
+
+#include "tests/c/array-direct.h"
+
+// IWYU: Struct is defined in...*array-indirect.h
+// IWYU: Struct needs a declaration
+void array(const struct Struct array[]);
```

Comment:
Since you just want an arbitrary struct type, you can use the reusable `struct Indirect` from `tests/c/direct.h`/`tests/c/indirect.h`.

I wrote up some design guidelines for tests: https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/IWYUStyle.md#regression-tests, they may be useful.